edx/cc2olx
# [BD-27] [BB-2680] Polish and prepare cc2olx converter

This PR lays ground for rework of https://github.com/openedx/cc2olx tool.

1. Added `setup.py` configuration, so package now can be installed and used like `cc2olx -i <FILE>`.
1. Added `tox` configuration, so all tests and linting can be run with `tox`.
1. Removed usage of raw `os.path` favoring `pathlib`.
1. Updated README.
1. Added `Makefile` with some utility commands.
1. Simplified `argparse` interface to have only `--inputs` argument, for both directories and files.
1. Reformatted code and removed some unused lines.
1. Removed `Cartridge.serialize` method. I'm not sure about it, but it seems that methods from `olx` module do the same.
1. Implemented zip output mode.
1. Covered mostly everything with tests.
1. Fixed one lti parsing error.

Since previous PR review:
1. Removed quotes replace from models file to make diff in this PR easier to read. I'll add reformat and quote style linting rule in following PR. 
1. It appeared, that `xml.dom.minidom` works differently for Python `3.5` and `3.8`, so I made some adjustments to fixtures.
1. `pathlib` support is much better in Python >= `3.6`, so I added a few type casts for backward compatibility with `3.5`.
1. All comments from previous PR are addressed.

**JIRA tickets**:
- [BLENDED-494](https://openedx.atlassian.net/browse/BLENDED-494)

**Discussions**: previous pull request with incorrect history: https://github.com/edx/cc2olx/pull/1 

**Dependencies**: None

**Testing instructions**:

1. Check that all tests and linting are passing:
    ```sh
    git clone https://github.com/edx/cc2olx
    cd cc2olx
    make test-all
    ```
   If your system lack needed python versions, you can use `pyenv` to install them, and then something like `pyenv local system 3.5.3 3.8.0`. After this `tox` should be able to see these versions. 

2. Check that tool converts some test `.imscc` files without errors: 
    ```sh
    virtualenv -p python3 .fenv
    . .fenv/bin/activate
    python setup.py install
    cc2olx -i test_data/allyworkshop.imscc
    ```

    After this, you should see new `output` directory in your current directory. It should contain `allyworkshop.tar.gz` file.

3. Open any course in studio, then click on `Tools` -> `Import` and try to import this `allyworkshop.tar.gz`. You should see that your course is replaced with new imported course.


**Author notes and concerns**:

1. I'm not sure what's the purpose of `zip` output mode, so I archive whole workspace directory.

**Reviewers**
- [ ] @stvstnfrd
